code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +678,24 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	if(takes_damage)
+		for(var/turf_to_check in RANGE_TURFS(1, loc))
+			if(isspaceturf(turf_to_check) && istype(loc))
+				var/is_open =  TRUE
+				for(var/obj/object_to_check in turf_to_check)
+					if(object_to_check.density == TRUE || object_to_check.CanAtmosPass == ATMOS_PASS_NO)
+						is_open = FALSE
+						break
+				if(is_open)
+					var/integrity = get_integrity()
+					if(integrity < 10)
+						damage += max((power * 0.004) * DAMAGE_INCREASE_MULTIPLIER, 0)
+					else if(integrity < 25)
+						damage += max((power * 0.002) * DAMAGE_INCREASE_MULTIPLIER, 0)
+					else if(integrity < 45)
+						damage += max((power * 0.005) * DAMAGE_INCREASE_MULTIPLIER, 0)
+					else if(integrity < 75)
+						damage += max((power * 0.0009) * DAMAGE_INCREASE_MULTIPLIER, 0)
```

Comment:
reminder, we want to make punching a hole in the sm a little less of a free pass, rather then making bombs the best way to delam. you can do this by making sure this is slower then a heat delam (which is capped at 2 damage a tick, you might want to move the codeblock in there and adjust the equations to match.)